ftp_blueprints/download_file.py

```diff
@@ -128,33 +133,35 @@ def determine_destination_name(
     return destination_name
 
 
-def find_ftp_file_names(client, prefix=''):
+def find_files_in_directory(
+        client,
+        folder_filter,
+        files,
+        folders):
     """
-    Fetched all the files in the folder on the FTP server
+    Pull in a list of all entities under a specific directory and categorize them into files and folders.
     """
-    try:
-        data = []
-        files = []
-        folders = []
-        client.dir(prefix, data.append)
-        for d in data:
-            if d.startswith('d'):
-                folders.append(d.split()[-1])
-            else:
-                name = d.split()[-1]
-                if prefix != '':
-                    files.append(f'{prefix}/{name}')
-                else:
-                    files.append(name)
-        for folder in folders:
-            if prefix:
-                folder = f'{prefix}/{folder}'
-            files.extend(find_ftp_file_names(client, folder))
-    except Exception as e:
-        print(f'Failed to find files in folder {prefix}')
-        raise(e)
+    original_dir = client.pwd()
+    names = client.nlst(folder_filter)
+    for name in names:
+        # Accounts for an issue where some FTP servers return file names
+        # without folder prefixes.
+        if '/' not in name:
+            name = f'{folder_filter}/{name}'
+
+        try:
+            client.cwd(name)
+            # If you can change the directory to the entity_name, it's a
+            # folder.
+            folders.append(f'{name}')
+        except ftplib.error_perm as e:
+            files.append(f'{name}')  # If you can't, it's a file.
```

Comment:
Is there a reason we're using the string format? Unless the `name` variable isn't a string, it can probably be dropped or swapped for `str()` instead.

Reply:
Ah. It used to be multiple variables combined, hence the f-string. You're right that we should just use `name` on its own at this point.